### PR TITLE
Updated ROS2 user guide: ROS2_Humble+PX4 simulation setup

### PR DESCRIPTION
**Pull request by nikhilsnayak3473**
- Upgraded the ROS2 user guide to the most recent LTS version of ROS2, namely Humble.
- Included comprehensive step-by-step instructions tailored to beginners, ensuring a straightforward and user-friendly setup experience.
- Additionally, incorporated a dedicated section guiding users through the installation process of QGroundControl, enabling them to seamlessly integrate it with the simulation environment.

### CODE REVIEW

**Comment by hamishwillee** [comment]
@nikhilsnayak3473 Thanks, but there were reasons I did this on Foxy and in this way.
1. Ubuntu 20.04 is a supported version of PX4 that does not run Humble
2. Ubuntu 22.04 does not run Gazebo Garden in our configuration, and the simulation functionality on PX4 is currently far more limited than Gazebo-Classic.
3. The headings say things like "Install ROS 2" rather than "Install ROS 2 Humble" because headings are external links, and by keeping this heading the same, the links still all work from elswhere.
4. I don't want to duplicate QGC instructions. Those vary from time to time and that would become a maintenance problem.
5. I would prefer not to duplicate the ROS2 instructions for similar reasons. One problem with spelling it out like this is that while it makes a good tutorial it does hide the fact that you don't have to install the desktop - you might well want to install the  other versions for a companion computer to make the installation more lightweight.

I would be happy enough with a Humble-centric set of instructions that also included foxy, but this would need to be done carefully. 

@beniaminopozzan I don't know if you feel the same way.

**Comment by nikhilsnayak3473** [comment]
Thank you, @hamishwillee, for your valuable comments and feedback. Considering that ROS Humble is the latest Long-Term Support (LTS) version and has support until 2027, it appears to be a prudent choice to transition to ROS Humble. In my opinion, beginners would likely prefer the most recent LTS version. I would greatly appreciate any suggestions or improvements that I can incorporate into my pull request. Once again, thank you for your input.

**Review by beniaminopozzan — COMMENTED** [review]
@hamishwillee , I agree that constraining the guide to ROS foxy is not great, especially since ROS foxy is going EOL. At the same time we are still missing CI on Ubuntu 22 which makes everything more difficult to maintain and test.

I think we should keep the instruction for foxy and add the ones for humble as soon as we have CI / a docker container for Ubuntu Jammy.

@hamishwillee , is it possible to have the switches to toggle between different option like the one used in the [ROS tutorials](http://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Cpp-Publisher-And-Subscriber.html#write-the-publisher-node) to toggle between instruction for Linux, Windows or MAC. We could use them to switch between Foxy and Humble.

File: en/ros/ros2_comm.md [hunk 1]
```diff
@@ -65,21 +65,52 @@ You only need to install PX4 if you need the simulator (as we do in this guide),
...
+cd
+sudo usermod -a -G dialout $USER
+sudo apt-get remove modemmanager -y
+sudo apt install gstreamer1.0-plugins-bad gstreamer1.0-libav gstreamer1.0-gl -y
+sudo apt install libqt5gui5 -y
+sudo apt install libfuse2 -y
+sudo apt install wget -y
+wget https://d176tv9ibo4jno.cloudfront.net/latest/QGroundControl.AppImage -P ~/QGroundControl.AppImage
+chmod +x ./QGroundControl.AppImage
+```
```
[the remark this answers is not in the document]
Comment (nikhilsnayak3473) [reply en/ros/ros2_comm.md:87]: Thank you for your comments, Fine I can modify it and add a link to QGC official docs for installation

File: en/ros/ros2_comm.md [hunk 2]
```diff
@@ -65,21 +65,52 @@ You only need to install PX4 if you need the simulator (as we do in this guide),
...
 Set up a PX4 development environment on Ubuntu in the normal way:
 
-1. [Setup the development environment for Ubuntu](../dev_setup/dev_env_linux_ubuntu.md)
-1. [Download PX4 source](../dev_setup/building_px4.md)
+```sh
+cd
+git clone https://github.com/PX4/PX4-Autopilot.git --recursive
+bash ./PX4-Autopilot/Tools/setup/ubuntu.sh
+cd PX4-Autopilot/
+make px4_sitl gz_x500
```
[the remark this answers is not in the document]
Comment (nikhilsnayak3473) [reply en/ros/ros2_comm.md:73]: Got it. Shall I add HEADLESS=1 flag for not starting the simulation

[the remark this answers is not in the document]
Comment (beniaminopozzan) [reply en/ros/ros2_comm.md:73]: That flag start the simulation without the GUI
To just compile you have to issue `make px4_sitl` without target.

[the remark this answers is not in the document]
Comment (nikhilsnayak3473) [reply en/ros/ros2_comm.md:73]: Got it. I will implement the changes and add another commit. If there are any other changes to be done please let me know

**Comment by hamishwillee** [comment]
THanks @nikhilsnayak3473 @beniaminopozzan - I will respond tomorrow when I am back in office.

**Comment by hamishwillee** [comment]
Thanks @beniaminopozzan @nikhilsnayak3473 

OK, I have subedited so now it supports both. I have chosen to allow the spelled out version that @nikhilsnayak3473 prefers, but in those cases I still link to originals. I removed QGC stuff - it is not needed for this tutorial.

@beniaminopozzan Is this OK.

Yes, I can make the docs support tabs, but if you want that, I will do it as a separate PR.

**Comment by nikhilsnayak3473** [comment]
Hello @hamishwillee and @beniaminopozzan,

I would also  like to contribute an offboard control example using Python for ROS Humble. I have thoroughly tested the example, and it closely resembles the current C++ example mentioned in the documentation.

**Comment by hamishwillee** [comment]
> I would also like to contribute an offboard control example using Python for ROS Humble. I have thoroughly tested the example, and it closely resembles the current C++ example mentioned in the documentation.

That would be very welcome.

**Comment by nikhilsnayak3473** [comment]
Where can I contribute it. Could you please tell me steps for submitting the PR for the example? @hamishwillee

**Comment by beniaminopozzan** [comment]
@nikhilsnayak3473 , you can start by updating the existing python offboard example node
https://github.com/PX4/px4_ros_com/tree/main/src/examples/offboard_py
and then document the example in the PX4 doc alongside the cpp example :)

At the same time, we should take into consideration other examples/tutorials, like
https://github.com/ARK-Electronics/ROS2_PX4_Offboard_Example
which is really well written! Do we want to duplicate the example just for the sake of having it in a PX4 repo or should it be enough to link the doc to it? @hamishwillee

**Comment by nikhilsnayak3473** [comment]
Dear @beniaminopozzan, 

I have tested the example you mentioned, but it appears to be incompatible with ROS2 Humble. Furthermore, I believe that replicating the functionality of the C++ code in Python and providing an explanation in the documentation would offer a better understanding and facilitate a good comparison between the two languages. The example repository you mentioned handles everything automatically and includes keyboard navigation. Instead, I suggest providing beginners with a simple takeoff and land example where they have to launch each process manually. This approach would be more convenient for their learning experience, in my opinion. Additionally, we can provide links to similar examples for users to explore and figure out on their own.

Thank you.

**Comment by hamishwillee** [comment]
I have updated to a tabbed version. The idea worked really well @beniaminopozzan. I am going to merge this and we can iterate as needed.

>> @nikhilsnayak3473 , you can start by updating the existing python offboard example node
https://github.com/PX4/px4_ros_com/tree/main/src/examples/offboard_py and then document the example in the PX4 doc alongside the cpp example :)

> I have tested the example you mentioned, but it appears to be incompatible with ROS2 Humble. Furthermore, I believe that replicating the functionality of the C++ code in Python and providing an explanation in the documentation would offer a better understanding and facilitate a good comparison between the two languages. The example repository you mentioned handles everything automatically and includes keyboard navigation. Instead, I suggest providing beginners with a simple takeoff and land example where they have to launch each process manually. This approach would be more convenient for their learning experience, in my opinion. Additionally, we can provide links to similar examples for users to explore and figure out on their own.

@nikhilsnayak3473 Yes, Ben is suggesting that the example needs to be updated for Humble. That's part of the argument.

I'm going to let you fight this out yourselves, but FWIW I slightly agree with @nikhilsnayak3473 here about having a simple starting point. Can we have both? The existing ROS example shows slightly different things. 

BUT, if we do have a new example that should also go in the same repo.

**Pull request merged**